unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/TextShape/Handler/ECSTextShapeComponentHandler.cs

```diff
@@ -0,0 +1,236 @@
+using System.Collections;
+using System.Collections.Generic;
+using DCL;
+using DCL.Controllers;
+using DCL.ECSComponents;
+using DCL.ECSRuntime;
+using DCL.Models;
+using TMPro;
+using UnityEngine;
+
+public class ECSTextShapeComponentHandler : IECSComponentHandler<PBTextShape>
+{
+    private const string BOTTOM = "bottom";
+    private const string TOP = "top";
+    private const string LEFT = "left";
+    private const string RIGHT = "right";
+    
+    private const string COMPONENT_NAME = "TextShape";
+
+    internal GameObject textGameObject;
+    internal TextMeshPro textComponent;
+    internal RectTransform rectTransform;
+    internal AssetPromise_Font promise;
+    
+    private PBTextShape currentModel;
+    private readonly DataStore_ECS7 dataStore;
+    private readonly AssetPromiseKeeper_Font fontPromiseKeeper;
+
+    public ECSTextShapeComponentHandler(DataStore_ECS7 dataStoreEcs7, AssetPromiseKeeper_Font fontPromiseKeeper)
+    {
+        dataStore = dataStoreEcs7;
+        this.fontPromiseKeeper = fontPromiseKeeper;
+    }
+
+    public void OnComponentCreated(IParcelScene scene, IDCLEntity entity)
+    {
+        textGameObject = new GameObject(COMPONENT_NAME);
+        textGameObject.AddComponent<MeshRenderer>();
+        rectTransform = textGameObject.AddComponent<RectTransform>();
+        textComponent = textGameObject.AddComponent<TextMeshPro>();
+        textGameObject.transform.SetParent(entity.gameObject.transform);
+        
+        textComponent.text = string.Empty;
+        
+        if (entity.meshRootGameObject == null)
+            entity.meshesInfo.meshRootGameObject = textGameObject;
+    }
+
+    public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+    {
+        RemoveModelFromPending(scene);
+        if (promise != null)
+            fontPromiseKeeper.Forget(promise);
+        GameObject.Destroy(textGameObject);
+        GameObject.Destroy(rectTransform);
+        GameObject.Destroy(textComponent);
```

Comment:
are those `Destroy` needed? aren't those components attached to `textGameObject`?